### PR TITLE
test: enable RenovateBot for java-shared-dependencies with check

### PR DESCRIPTION
**Pull request by suztomo**
- Enable RenovateBot for java-shared-dependencies module
- Add check from original repository https://github.com/googleapis/java-shared-dependencies/blob/main/.github/workflows/version-check.yaml
- Run integration tests for all modules when java-shared-dependencies is touched (generate_modified_modules_list in common.sh)

### CODE REVIEW

File: .github/workflows/shared_dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - uses: stCarolas/setup-maven@v4
+      with:
+        maven-version: 3.8.1
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - run: java -version
+    - name: Install java-core (part of java-shared-dependencies)
```
Comment (suztomo) [inline .github/workflows/shared_dependencies.yaml:23]:
```
Run mvn -B -V -ntp install -DskipTests --also-make -f java-core
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /opt/hostedtoolcache/maven/3.8.1/x64
Java version: 11.0.18, vendor: Azul Systems, Inc., runtime: /opt/hostedtoolcache/jdk/11.0.18/x64
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-1031-azure", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
Error: ] Some problems were encountered while processing the POMs:
Error:  Non-resolvable import POM: Could not find artifact com.google.cloud:google-cloud-shared-dependencies:pom:3.1.3-SNAPSHOT in google-maven-central-copy (https://maven-central.storage-download.googleapis.com/maven2) @ com.google.cloud:google-cloud-jar-parent:1.4.0-SNAPSHOT, /home/runner/work/google-cloud-java/google-cloud-java/google-cloud-jar-parent/pom.xml, line 27, column 19
Error:  'dependencies.dependency.version' for javax.annotation:javax.annotation-api:jar is missing. @ com.google.cloud:google-cloud-jar-parent:1.4.0-SNAPSHOT, /home/runner/work/google-cloud-java/google-cloud-java/google-cloud-jar-parent/pom.xml, line 1[26](https://github.com/googleapis/google-cloud-java/actions/runs/4066699468/jobs/7003119762#step:6:27), column 21
```

**Comment by suztomo** [comment]
Good. It detected the perfmark upper-bound failure:

```
Error:  Rule 2: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for io.perfmark:perfmark-api:0.25.0 [runtime] paths to dependency are:
+-com.google.cloud:shared-dependencies-upper-bound-test:3.1.3-SNAPSHOT
  +-io.grpc:grpc-core:1.52.1
    +-io.perfmark:perfmark-api:0.25.0 [runtime]
and
+-com.google.cloud:shared-dependencies-upper-bound-test:3.1.3-SNAPSHOT
  +-io.grpc:grpc-netty:1.52.1
    +-io.perfmark:perfmark-api:0.25.0 [runtime]
and
+-com.google.cloud:shared-dependencies-upper-bound-test:3.1.3-SNAPSHOT
  +-io.grpc:grpc-netty-shaded:1.52.1
    +-io.perfmark:perfmark-api:0.25.0 [runtime]
and
+-com.google.cloud:shared-dependencies-upper-bound-test:3.1.3-SNAPSHOT
  +-io.grpc:grpc-okhttp:1.52.1
    +-io.perfmark:perfmark-api:0.25.0 [runtime]
and
+-com.google.cloud:shared-dependencies-upper-bound-test:3.1.3-SNAPSHOT
  +-com.google.cloud:google-cloud-core-grpc:2.9.5-SNAPSHOT
    +-io.perfmark:perfmark-api:0.26.0 [runtime]
]
```

File: .github/workflows/shared_dependencies.yaml (hunk 1 above)
Comment (suztomo) [reply .github/workflows/shared_dependencies.yaml]: Resolved.

**Review by meltsufin — APPROVED** [review]

File: .kokoro/common.sh [hunk 2]
```diff
@@ -102,7 +102,8 @@ function generate_modified_modules_list() {
   modified_module_list=()
   # If either parent pom.xml is touched, run ITs on all the modules
   parent_pom_modified=$(echo "${modified_files}" | grep -E '^google-cloud-(pom|jar)-parent/pom.xml$' || true)
-  if [[ (-n $parent_pom_modified) || ("${TEST_ALL_MODULES}" == "true") ]]; then
+  shared_dependencies_modified=$(echo "${modified_files}" | grep -E '^java-shared-dependencies' || true)
+  if [[ (-n $parent_pom_modified) || (-n $shared_dependencies_modified) || ("${TEST_ALL_MODULES}" == "true") ]]; then
     modified_module_list=(${maven_modules[*]})
     echo "Testing the entire monorepo"
```
Comment (suztomo) [inline .kokoro/common.sh:108]: This logic is working.

Screenshot 2023-02-01 at 12 24 15 PM

**Comment by suztomo** [comment]
`com.google.cloud.bigqueryconnection.v1.it.ITSystemTest.testListConnections` failed. Retrying.

```
com.google.api.gax.rpc.UnimplementedException:
io.grpc.StatusRuntimeException: UNIMPLEMENTED: HTTP status code 404
invalid content-type: text/html; charset=UTF-8
headers: Metadata(:status=404,date=Wed, 01 Feb 2023 17:45:28 GMT,content-type=text/html; charset=UTF-8,server=scaffolding on HTTPServer2,x-xss-protection=0,x-frame-options=SAMEORIGIN,content-length=1630)
DATA-----------------------------
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//[www.google.com/images/errors/robot.png](http://www.google.com/images/errors/robot.png)) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//[www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png](http://www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png)) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//[www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png](http://www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png)) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//[www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png](http://www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png)) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//[www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png](http://www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png)) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//[www.google.com/](http://www.google.com/)><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/google.cloud.bigquery.connection.v1.ConnectionService/ListConnections</code> was not found on this server.  <ins>That’s all we know.</ins>

	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:108)
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:41)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:86)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:66)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:67)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1132)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1270)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1038)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:808)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:574)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:544)
	at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)
	at io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)
	at com.google.api.gax.grpc.ChannelPool$ReleasingClientCall$1.onClose(ChannelPool.java:541)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:576)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:757)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:736)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
		at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
		at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
		at com.google.cloud.bigqueryconnection.v1.ConnectionServiceClient.listConnections(ConnectionServiceClient.java:536)
		at com.google.cloud.bigqueryconnection.v1.it.ITSystemTest.testListConnections(ITSystemTest.java:124)
```

**Comment by suztomo** [comment]
The BigQueryConnection test failed 2 times in a row. Retrying 3rd time.

**Comment by suztomo** [comment]
Retrying 4th time.

**Pull request merged**